core/main_loop.py

```diff
@@ -218,6 +218,8 @@ def Interactive(flag, cmd_ev, c_parser, display, prompt_plugin, waiter, errfmt):
                         # POSIX shell behavior: waitpid(-1) and show job "Done"
                         # messages
                         r = waiter.WaitForOne(waitpid_options=WNOHANG)
+                        while r == process.W1_OK:
+                            r = waiter.WaitForOne(waitpid_options=WNOHANG)
```

Comment:
Should we place a limit on this loop to avoid accidentally starving the outer loop? A storm of process state changes could make the shell feel sluggish for a moment (it would have to be a pretty big storm, though...)